### ISSUE TITLE
please add donate button

### ISSUE THREAD

**Issue by dzmitry-lahoda** [oxalica/nocargo#12]
in crypto preferrably

**Comment by oxalica** [comment]
I'm not quite knowing how many demands for this project are, also the number of users. So I'm recently not active on this project.
At first it's just a proof of concept (PoC) for non-FOD and non-generating crate-granularity builder.

Please let me know in which scene are you using this project.

If you do want to donate, there are cryptocurrency addresses in the folded section of [my GitHub profile page](https://github.com/oxalica).

**Comment by cor** [comment]
We'd love to use this for our project in the cryptocurrency scene, as crate-level binary caching would speed up our builds tremendously

**Comment by dzmitry-lahoda** [comment]
polkadot is crazy about feature usage (abuse) also they have built their own stuff for reproducible builds and rust toml patching. but we want more - want other rust chains and not only rust.  and universally handle all things (not only rust - e.g. rust depends on protobuf or go).

we already use rustPlatform/crane, but these are limiting.

**Comment by gmarti** [comment]
Using crane here, but I tested nocargo and would like to actually use it for our builds

Missing some features here and there, would be glad to contribute when i have more time

**Comment by KaiserKarel** [comment]
Same, we'd switch entirely away from Cargo for this :)

**Comment by skeet70** [comment]
[IronCore](https://github.com/IronCoreLabs) would also love to use this. In the cryptography (not blockchain/currency) and privacy space. We're looking at `cargo2nix` to speed up CI builds, but struggle with having to manage the duplicated information between `Cargo.nix` and `Cargo.lock`. This project seems great to avoid that, and I tested it locally and covered a lot of the bases already. We do have developers on M1 Macs though, and do release builds in CI on/for M1 and x86_64 Macs. For our use case the two missing features are:

* arm64 and x86_64 darwin support
* the `test` target, since most of the CI build time is for `cargo test` right now

I'll try to encourage some of the devs on Macs to help out with the open Mac support PR at the next hackathon. I'm not sure what's involved with supporting `test`.

**Comment by oxalica** [comment]
> * arm64 and x86_64 darwin support

I don't have these devices so it's hard to test for me. PRs are welcome.

> * the `test` target, since most of the CI build time is for `cargo test` right now

It should work alike `bin` targets, except the compile flags.

Hopefully I'll pick up this project again when I'm less busy. cargo also added many features in the past months.

**Comment by skeet70** [comment]
> > * arm64 and x86_64 darwin support
> 
> I don't have these devices so it's hard to test for me. PRs are welcome.
> 
> > * the `test` target, since most of the CI build time is for `cargo test` right now
> 
> It should work alike `bin` targets, except the compile flags.
> 
> Hopefully I'll pick up this project again when I'm less busy. cargo also added many features in the past months.

Yep, my comment was in response to your ask for use cases. I made an attempt at getting `test` working, but I'm a Rust person and not super comfortable with complex bash or nix so I didn't make it very far. I'll give it another shot in the future. I also directed our Mac owning folks to the draft PR for Mac support, hopefully one of them picks it up.
